pkg/storage/redis/oidctxstore/tx_store.go

```diff
@@ -0,0 +1,64 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package oidctxstore
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/go-redis/redis/v8"
+)
+
+// TxStore stores oidc transactions in redis.
+type TxStore struct {
+	client    *redis.Client
+	opTimeout time.Duration
+}
+
+// New creates TxStore.
+func New(client *redis.Client, opTimeout time.Duration) *TxStore {
+	return &TxStore{
+		client:    client,
+		opTimeout: opTimeout,
+	}
+}
+
+// GetAndDelete get and then delete transaction by one time token.
+func (ts *TxStore) GetAndDelete(key string) ([]byte, bool, error) {
```

Comment:
what is the use of this function?

Reply:
Atomically get transaction by one-time token. after this, it will be deleted from the store. The method is used in the end stage of interaction. To mitigate replay attacks.